### PR TITLE
Add 2/2/2 role limit

### PR DESCRIPTION
**Pull request by novalderya**
Add 2-2-2 role lock with doomfist as dps
role_lock_sup
role_lock_dps
role_lock_tank

Only tested with bots

### CODE REVIEW

**Comment by snappycreeper** [comment]
just curious, is there a way to change role?

**Comment by novalderya** [comment]
> just curious, is there a way to change role?

I don't think in one round because if everybody choose their role, there will be 2 tank for example and so no one could take a tank since the number equal 2, but in like switching side I think you can change at this moment.

**Comment by snappycreeper** [comment]
hm i think a way to switch role with someone else should be added

**Comment by MaxwellJung** [comment]
> hm i think a way to switch role with someone else should be added

That would be nice, but I think this PR is just recreating the 2/2/2 role lock enforcement native to this game.

File: src/ow1/role_limit.opy [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#!mainFile "../main.opy"
+
+#Global variables
+
+globalvar HeroListDamage 0
+globalvar HeroListTank 1
+globalvar HeroListSupport 2
+globalvar RoleLimitDamage 3
+globalvar RoleLimitTank 4
+globalvar RoleLimitSupport 5
```
Comment (MaxwellJung) [inline src/ow1/role_limit.opy:10]: Change to snake_case and remove the ordering numbers (0, 1, 2, ...)

File: src/ow1/role_limit.opy [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    HeroListSupport.remove([Hero.KIRIKO,Hero.LIFEWEAVER])
+    RoleLimitDamage = 2
+    RoleLimitTank = 2
+    RoleLimitSupport = 2
+
+
+rule "[role_limit.opy]: Role limit lockout damage":
+    @Event eachPlayer
+    @Condition len([player for player in getPlayers(eventPlayer.getTeam()) if player.getCurrentHero() in HeroListDamage]) >= RoleLimitDamage
+    @Condition eventPlayer.getCurrentHero() in HeroListDamage == false
```
Comment (MaxwellJung) [inline src/ow1/role_limit.opy:28]:
```suggestion
    @Condition (eventPlayer.getCurrentHero() in HeroListDamage) == false
```

Add parenthesis for safety and clarify; I forgot parenthesis in some condition in another code that took hours to debug.

File: src/ow1/role_limit.opy [hunk 3]
```diff
@@ -0,0 +1,69 @@
+#!mainFile "../main.opy"
+
+#Global variables
```
Comment (MaxwellJung) [inline src/ow1/role_limit.opy:3]: If you sourced this code from someone else, make sure you credit them with comments starting at line 3 (below the mainfile declarative)

File: src/ow1/role_limit.opy [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    eventPlayer.setAllowedHeroes(eventPlayer.getAllowedHeroes().exclude(HeroListSupport))
+
+
+rule "[role_limit.opy]: Role limit enable support":
+    @Event eachPlayer
+    @Condition len([player for player in getPlayers(eventPlayer.getTeam()) if player.getCurrentHero() in HeroListSupport]) < RoleLimitSupport
+    
+    eventPlayer.setAllowedHeroes(eventPlayer.getAllowedHeroes().concat(HeroListSupport))
+
+
```
Comment (MaxwellJung) [inline src/ow1/role_limit.opy:69]:
```suggestion
```

**Review by MaxwellJung — CHANGES_REQUESTED** [review]

File: src/ow1/role_limit.opy (hunk 4 above)
Comment (MaxwellJung) [reply src/ow1/role_limit.opy:69]: remove extra empty line

File: src/ow1/role_limit.opy (hunk 2 above)
Comment (novalderya) [reply src/ow1/role_limit.opy]: Done

**Review by MaxwellJung — APPROVED** [review]
I'll merge it to staging, but expect this PR to be reverted if it doesn't work in staging.

**Pull request merged**